lightning-graph-sync/README.md

```diff
@@ -0,0 +1,90 @@
+# lightning-graph-sync
+
+This crate exposes functionality for rapid gossip graph syncing, aimed primarily
+at mobile clients.
+
+## Mechanism
+
+The (presumed) server sends a compressed gossip response containing gossip data.
+The gossip data is formatted compactly, omitting signatures and opportunistically
+incremental.
+
+Essentially, the serialization structure is as follows:
+
+1. Fixed prefix bytes `76, 68, 75, 1` (the first three bytes are ASCII for `LDK`)
+    - The purpose of this prefix is to identify the serialization format, should other
+    rapid gossip sync formats arise in the future.
+    - The fourth byte is the protocol version in case our format gets updated
+2. A `CompactSize` indicating the number of channel announcement messages to follow
+3. `[CustomChannelAnnouncement]` (array of significantly stripped down channel announcements)
+4. A `CompactSize` indicating the number of channel update messages to follow
+5. `[CustomChannelUpdate]`
+
+You will also notice that `NodeAnnouncement` messages are skipped altogether.
+
+The data is then applied to the current network graph, artifically backdated 7 days from the current time to
+make sure more recent updates obtained directly from gossip are not accidentally overwritten.
+
+### CustomChannelAnnouncements
+
+To achieve compactness and avoid data repetition, we're sending a significantly stripped down version of the
+channel announcement message, which contains only the following data:
+
+1. `channel_features`: `u16` + `n`, where `n` is the number of bytes indicated by the first `u16`
+2. `short_channel_id`: `u64`
+3. `node_id_1`: `ECPoint`
+4. `node_id_2`: `ECPoint`
+
+### CustomChannelUpdate
+
+For the purpose of rapid syncing, we have deviated from the channel update format
+specified in BOLT 7 significantly. Our custom channel updates are structured as follows:
+
+1. `short_channel_id`: `u64`
+2. `custom_channel_flags`: `u8`
+3. `update_data`
+
+Specifically, our custom channel flags break down like this:
+
+| 128                 | 64 | 32 | 16 | 8 | 4 | 2                | 1         |
+|---------------------|----|----|----|---|---|------------------|-----------|
+| Incremental update? |    |    |    |   |   | Disable channel? | Direction |
+
+If the most significant bit is set to `0`, indicating a non-incremental update, 
```

Comment:
nit: whitespace at EOL here and a few places. local `git show` should highlight them.

Reply:
I thought it's considered best practice to have a space at the end of a word prior to a newline, but I can remove it

Reply:
Hmm, I've never heard of that, and my almost-default vim will automagically remove them when it line-wraps.